cmd/launcher/desktop/desktop.go

```diff
@@ -39,23 +41,22 @@ func handleSignals() {
 
 // continuously monitor for ppid and exit if parent process terminates
 func exitWhenParentGone() {
-	ticker := time.NewTicker(2 * time.Second)
-	defer ticker.Stop()
-
-	f := func() {
-		if os.Getppid() <= 1 {
-			fmt.Println("parent process is gone, exiting")
-			systray.Quit()
-			os.Exit(1)
-		}
-	}
+	for ; true; <-time.NewTicker(2 * time.Second).C {
+		ppid := os.Getppid()
 
-	f()
+		if ppid <= 1 {
+			break
+		}
 
-	for {
-		select {
-		case <-ticker.C:
-			f()
+		ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+		exists, err := process.PidExistsWithContext(ctx, int32(ppid))
+		cancel()
+		if err != nil || !exists {
+			break
 		}
 	}
+
+	fmt.Println("parent process is gone, exiting")
```

Comment:
println :|

Reply:
you prefer `fmt.Print("\nparent process is gone, exiting")` or we need a logger?

Reply:
I feel like we should figure out the logger before GA.

Reply:
But we can not block this PR merge on it.